frontend/src/screens/dashboard/docsqa/main/index.tsx

```diff
@@ -0,0 +1,388 @@
+      setIsRunningPrompt(false)
+      notifyError('Failed to retrieve answer', err)
+    }
+  }
+
+  const createChatApplication = async (
+    applicationName: string,
+    questions: [],
+    setApplicationName: (name: string) => void,
+  ) => {
+    if (!applicationName) {
+      return notify('error', 'Application name is required')
+    }
+    const selectedModel = allEnabledModels.find(
+      (model: any) => model.name == selectedQueryModel,
+    )
+
+    try {
+      await createApplication({
+        name: `${applicationName}-rag-app`,
+        config: {
+          collection_name: selectedCollection,
+          model_configuration: {
+            name: selectedModel.name,
+            provider: selectedModel.provider,
+            ...JSON.parse(modelConfig),
+          },
+          retriever_name: selectedRetriever?.name ?? '',
+          retriever_config: JSON.parse(retrieverConfig),
+          prompt_template: promptTemplate,
+          query_controller: selectedQueryController,
+        },
+        questions,
+      }).unwrap()
+      setApplicationName('')
+      setIsCreateApplicationModalOpen(false)
+      notify('success', 'Application created successfully')
+    } catch (err: any) {
+      notify('error', 'Failed to create application', err?.data?.detail)
+    }
+  }
+
+  const resetQA = () => {
+    setAnswer('')
+    setErrorMessage(false)
+    setPrompt('')
+  }
+
+  useEffect(() => {
+    if (collections && collections.length) setSelectedCollection(collections[0])
+
+    if (allQueryControllers && allQueryControllers.length)
+      setSelectedQueryController(allQueryControllers[0])
+
+    if (allEnabledModels && allEnabledModels.length)
+      setSelectedQueryModel(allEnabledModels[0].name)
+
+    if (allRetrieverOptions && allRetrieverOptions.length) {
+      setSelectedRetriever(allRetrieverOptions[0])
+      setPromptTemplate(allRetrieverOptions[0].promptTemplate)
+    }
+    if (selectedRetriever)
+      setRetrieverConfig(JSON.stringify(selectedRetriever.config, null, 2))
+  }, [
+    collections,
+    allQueryControllers,
+    allEnabledModels,
+    allRetrieverOptions,
+    selectedRetriever,
+  ])
+
+  return (
+    <>
+      {isCreateApplicationModalOpen && (
+        <ApplicationModal
+          createChatApplication={createChatApplication}
+          isCreateApplicationLoading={isCreateApplicationLoading}
+          isCreateApplicationModalOpen={isCreateApplicationModalOpen}
+          setIsCreateApplicationModalOpen={setIsCreateApplicationModalOpen}
+        />
+      )}
+      <div className="flex gap-5 h-[calc(100vh-6.5rem)] w-full">
+        {isCollectionsLoading ? (
+          <div className="h-full w-full flex items-center">
+            <Spinner center big />
+          </div>
+        ) : selectedCollection ? (
+          <>
+            <div className="h-full border rounded-lg border-[#CEE0F8] w-[23.75rem] bg-white p-4 overflow-auto">
+              <div className="flex flex-col gap-3">
```

Comment:
I suppose this can also be broken down into `<Sidebar /> `or `<QueryConfig />`

Reply:
Yes i was also thinking same but in that case will have to do more prop drilling like sidebar have all config data and right section will have that form so we need to supply it to both so either we can pass from parent so that will be a lot of props for both file if it is fine i can do like that